doc/self-hosted/build.md

```diff
@@ -0,0 +1,60 @@
+# Build instructions for code-server 🏗️
+
+code-server has several yarn tasks for building the entire binary or specific portions of the app. This guide provides steps to do the aforementioned tasks. Coder recommends having Node 10.15.1+ installed.
+
+**NOTE:** Windows compatibility is in the works 
+> Yarn installation: https://yarnpkg.com/lang/en/docs/install
+
+### Build the binary
+1. Clone the [repository](https://github.com/codercom/code-server.git)
+2. `cd` into `code-server`
+3. Run `yarn`
+4. Run `yarn task build:server:binary`
+   > Binary will be placed in `packages/server` named after the platform and architecture (`cli-darwin-x64`)
+
+### Run without building
+1. Clone the [repository](https://github.com/codercom/code-server.git)
+2. `cd` into `code-server`
+3. Run `yarn`
+4. Run `yarn start`
```

Comment:
Currently you have to run a few other steps. I think at minimum you have to build the bootstrap fork, possibly some other things. We could build them as part of `yarn start` though.

We might want to change `yarn start` to include `--no-auth` in the `package.json`. The problem is that `webpack-dev-server` is handling the requests, not `express`, so it doesn't show the login screen and just directly serves the IDE bundle. But without authentication the websocket requests won't work. 

We can probably refactor a bit to make authentication work though. I think we'll want to make it so everything automatically reloads during development while we're at it.